store/sqlstore/post_store.go

```diff
@@ -2440,9 +2432,10 @@ func (s *SqlPostStore) updateThreadsFromPosts(transaction *gorp.Transaction, pos
 			// metadata exists, update it
 			for _, post := range posts {
 				thread.ReplyCount += 1
-				if !thread.Participants.Contains(post.UserId) {
-					thread.Participants = append(thread.Participants, post.UserId)
+				if thread.Participants.Contains(post.UserId) {
+					thread.Participants = thread.Participants.Remove(post.UserId)
 				}
+				thread.Participants = append(thread.Participants, post.UserId)
```

Comment:
Shouldn't this be in an else clause? If we are always unconditionally adding the userid, then it makes the removing part moot.

Reply:
@agnivade we keep de-duplicated user ids in the participants so when a user replies again the id gets added at the end and the previous value is deleted.

Reply:
Ahh I see now. The participants slice is now order-sensitive.

I don't have much knowledge of CRT, but in general, I would suggest to do it like we are doing it for posts. i.e. keep the raw data and the order as separate items. I'll leave it to your better judgement.

Reply:
That seems pretty reasonable, but then we would need to scan the posts data each time we fetch the thread(s).

Reply:
No worries from my end. Once you add a test, it's good to go from my side. :)